airbyte-analytics/src/main/java/io/airbyte/analytics/SegmentTrackingClient.java

```diff
@@ -68,15 +89,17 @@ public void identify(final UUID workspaceId) {
       identityMetadata.put(AIRBYTE_ROLE, airbyteRole);
     }
 
+    final String joinKey = trackingIdentity.getCustomerId().toString();
     analytics.enqueue(IdentifyMessage.builder()
         // user id is scoped by workspace. there is no cross-workspace tracking.
-        .userId(trackingIdentity.getCustomerId().toString())
+        .userId(joinKey)
         .traits(identityMetadata));
   }
 
   @Override
   public void alias(final UUID workspaceId, final String previousCustomerId) {
-    analytics.enqueue(AliasMessage.builder(previousCustomerId).userId(identityFetcher.apply(workspaceId).getCustomerId().toString()));
+    final var joinKey = identityFetcher.apply(workspaceId).getCustomerId().toString();
```

Comment:
if the tracking client hasn't been initialized can customerId be null? should we handle that case while we're refactoring? if not obvious skip it.

Reply:
Hmmm.. if this is null, this would be unexpected and should scream loudly to us. Because of this, I am inclined to not catch the NPE and leave it as is. WDYT?